packages/core/src/fields/types/text/index.ts

```diff
@@ -24,7 +27,7 @@ export type TextFieldConfig<ListTypeInfo extends BaseListTypeInfo> =
       match?: { regex: RegExp, explanation?: string }
       length?: { min?: number, max?: number }
     }
-    defaultValue?: string
+    defaultValue?: string | null
```

Comment:
Unlike other fields, the ambiguity of `?:` representing `''` or `null` is why this needs to exist.
We should allow `null` as a `defaultValue` typically, and maybe check it statically too.